t/t2040-checkout-symlink-attr.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+
+test_description='checkout symlinks with `symlink` attribute on Windows
+
+Ensures that Git for Windows creates symlinks of the right type,
+as specified by the `symlink` attribute in `.gitattributes`.'
+
+# Tell MSYS to create native symlinks. Without this flag test-lib's
+# prerequisite detection for SYMLINKS doesn't detect the right thing.
+MSYS=winsymlinks:nativestrict && export MSYS
+
+. ./test-lib.sh
+
+if ! test_have_prereq MINGW,SYMLINKS
+then
+	skip_all='skipping $0: MinGW-only test, which requires symlink support.'
+	test_done
+fi
+
+# Adds a symlink to the index without clobbering the work tree.
+cache_symlink () {
+	sha=$(printf '%s' "$1" | git hash-object --stdin -w) &&
+	git update-index --add --cacheinfo 120000 $sha "$2"
+}
+
+# MSYS2 is very forgiving, it will resolve symlinks even if the
+# symlink type isn't correct. To make this test meaningful, try
+# them with a native, non-MSYS executable.
+cat_native () {
+	filename=$(cygpath -w "$1") &&
+	cmd.exe /c "type \"$filename\""
+}
+
+test_expect_success 'checkout symlinks with attr' '
+	cache_symlink file1 file-link &&
+	cache_symlink dir dir-link &&
+
+	(
+		echo "file-link symlink=file" &&
+		echo "dir-link symlink=dir"
+	) >.gitattributes &&
```

Comment:
How about

```
	printf "file-link symlink=file\ndir-link symlink=dir\n" >.gitattributes &&
```

instead?

Reply:
This stanza was copied from https://github.com/git-for-windows/git/blob/d4f95fc6d2b604c7f66a0420e722588f85a043bc/t/t0003-attributes.sh#L48-L51

But I don't care actually, so I've applied your suggestion.

Reply:
Yep. Sadly, not all parts of our test suite may serve as stellar examples to follow.